src/Aspire.Hosting.PostgreSQL/PostgresDatabaseResource.cs

```diff
@@ -23,9 +24,18 @@ public class PostgresDatabaseResource(string name, string databaseName, Postgres
     /// <summary>
     /// Gets the connection string expression for the Postgres database.
     /// </summary>
-    public ReferenceExpression ConnectionStringExpression =>
-       ReferenceExpression.Create($"{Parent};Database={DatabaseName}");
+    public ReferenceExpression ConnectionStringExpression
+    {
+        get
+        {
+            var connectionStringBuilder = new DbConnectionStringBuilder
+            {
+                ["Database"] = DatabaseName
+            };
```

Comment:
What's special about this that we need to use the builder? Does it escape properly?

Reply:
It does, and there is a functional test to validate we can pass database name with special chars to ensure the connection string and the default CREATE DATABASE script are correctly escaping.

Reply:
I did it for Sql Server too btw.